terraform/redis/cluster/provision/data.tf

```diff
@@ -19,6 +19,10 @@ data "aws_vpc" "vpc" {
 
 locals {
   instance_types = {
+    // The now deprecated property cache_size has a new default value of -1
+    // which we are mapping to the empty string. This way we minimise code changes
+    // while still can detect when node_type ends up being empty or unspecified
+    -1 = ""
```

Comment:
question: if `cache_size` is `-1` and `node_type` is `""` and because it's a computed property in `terraform-provider-aws`, the final `node_type` will be the default value selected by the provider. Is this what we want?

Reply:
The provider doesn't have a default value for `node_type`.
If we are concerned about this changing inadvertendly in the future, then we need to either:
1. Rethink the user_inputs logic
2. Set "-1" value to an invalid string.

I believe I like option "2." and we could add it to this PR.
- It is an easy change
- It can be used to provide good error message by setting the value to be "you need specify a value for node_type"
- It is future-proof (not risk of accepting the default).

I will implement option 2 unless you have a different opinion.

Reply:
Removed in commit: https://github.com/cloudfoundry/csb-brokerpak-aws/pull/774/commits/e1b98c701d764d72ef155faf667adf7868ceb771